src/librustdoc/html/render/print_item.rs

```diff
@@ -143,8 +139,7 @@ pub(super) fn print_item(
         src_href: src_href.as_deref(),
     };
 
-    let teractx = tera::Context::from_serialize(item_vars).unwrap();
-    let heading = templates.render("print_item.html", &teractx).unwrap();
+    let heading = item_vars.render().unwrap();
```

Comment:
A further performance improvement here would be to make sure that `Buffer` impls `core::fmt::Write`, which would avoid allocating a temporary `String` here. `Buffer` internally seems to just hold a `String` and it has `write_str()` and `write_fmt()` methods already, so perhaps it would make sense to implement that trait? I'm not sure how big/relevant the performance improvement would be.